f5_openstack_agent/lbaasv2/drivers/bigip/disconnected_service.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2016 F5 Networks Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from eventlet import greenthread
+from f5_openstack_agent.lbaasv2.drivers.bigip.network_helper import \
+    NetworkHelper
+from neutron.plugins.common import constants as plugin_const
+from neutron_lbaas.services.loadbalancer import constants as lb_const
+from oslo_log import helpers as log_helpers
+from oslo_log import log as logging
+from oslo_utils import timeutils
+
+LOG = logging.getLogger(__name__)
+
+
+class DisconnectedServicePolling(object):
+    def __init__(self, driver):
+        self.driver = driver
+        self.enabled = driver.conf.f5_network_segment_physical_network
+        if self.enabled:
+            greenthread.spawn(self.polling_thread)
+        self.start_time = {}
+
+    def scan(self):
+        """Periodically scan for disconnected virtual servers.
+
+        :return: Return the number of seconds to wait before invoking this
+            method again.
+        """
+
+        LOG.debug("scanning disconnected networks")
+        # find active loadbalancers for this agent to see which virtual
+        # servers have become un-disconnected.
+        d = self.driver
+        if d.plugin_rpc:
+            loadbalancer_ids = d.plugin_rpc.get_all_loadbalancers()
+            for lb in loadbalancer_ids:
+                id = lb['lb_id']
+                service = d.plugin_rpc.get_service_by_loadbalancer_id(id)
+                listeners = service.get('listeners', None)
+                service.pop('listeners', None)
+                provisioning_status = \
+                    service['loadbalancer']['provisioning_status']
+                if not d.disconnected_service.is_service_connected(service):
+                    if id in self.start_time:
+                        if timeutils.is_older_than(
+                                self.start_time[id],
+                                d.conf.f5_network_segment_gross_timeout):
+                            LOG.error("failed to connect loadbalancer %s to "
```

Comment:
Should this log message be worded so that we know we hit the timeout versus some other error conditions?